bindings_ffi/src/lib.rs

```diff
@@ -76,25 +78,167 @@ pub async fn create_client(
         xmtp_client.wallet_address()
     );
     Ok(Arc::new(FfiXmtpClient {
-        inner_client: xmtp_client,
+        inner_client: Arc::new(xmtp_client),
     }))
 }
 
 #[derive(uniffi::Object)]
 pub struct FfiXmtpClient {
-    inner_client: RustXmtpClient,
+    inner_client: Arc<RustXmtpClient>,
 }
 
 #[uniffi::export]
 impl FfiXmtpClient {
     pub fn wallet_address(&self) -> Address {
         self.inner_client.wallet_address()
     }
+
+    pub fn conversations(&self) -> Arc<FfiConversations> {
+        Arc::new(FfiConversations {
+            inner_client: self.inner_client.clone(),
+        })
+    }
+}
+
+#[derive(uniffi::Object)]
+pub struct FfiConversations {
+    inner_client: Arc<RustXmtpClient>,
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl FfiConversations {
+    pub async fn new_conversation(
+        &self,
+        wallet_address: String,
+    ) -> Result<Arc<FfiConversation>, GenericError> {
+        let conversations = xmtp::conversations::Conversations::new(self.inner_client.as_ref());
+        let convo = conversations
+            .new_secret_conversation(wallet_address)
+            .map_err(|e| e.to_string())?;
+        // TODO: This should happen as part of `new_secret_conversation` and should only send to new participants
+        convo.initialize().await.map_err(|e| e.to_string())?;
+
+        let out = Arc::new(FfiConversation {
+            inner_client: self.inner_client.clone(),
+            peer_address: convo.peer_address(),
+        });
+
+        Ok(out)
+    }
+
+    pub async fn list(&self) -> Result<Vec<Arc<FfiConversation>>, GenericError> {
+        let inner = self.inner_client.as_ref();
+        let conversations = xmtp::conversations::Conversations::new(inner);
```

Comment:
The lifetimes on our `Conversation` and `Conversations` method are incompatible with Uniffi exported objects, so instead we never store the structs on anything that gets exposed and just build them on demand. That sidesteps the need for compatible lifetimes with objects that move over the FFI boundary.